zilliqa/src/api/eth.rs

```diff
@@ -273,6 +276,119 @@ fn get_block_transaction_count_by_number(
     Ok(block.map(|b| b.transactions.len().to_hex()))
 }
 
+#[derive(Deserialize, Default)]
+#[serde(default, rename_all = "camelCase")]
+struct GetLogsParams {
+    from_block: Option<BlockNumber>,
+    to_block: Option<BlockNumber>,
+    address: Option<OneOrMany<H160>>,
```

Comment:
With `Option<OneOrMany<>>`, is there a meaningful difference between `None` and `Some(Null)`? Otherwise the `Option<>` seems superfluous here (unless this is necessary for param parsing)

Reply:
There is a difference (though maybe not a useful one)
https://github.com/Zilliqa/zq2/blob/3655674d671cf0b70bb7c2a9c0e652446ec8bf1a/zilliqa/src/api/eth.rs#L360-L364

When `params.address = None`, this returns `true`. This means the log in question is always included.

When `params.address = Some(Null)`, this returns `false` because `OneOrMany::Null.contains(x) == false` for all `x`. This means the log in question is always excluded.

So passing `null` for the `address` will always result in an empty response, while passing nothing will do no extra filtering. Arguably we could return an error in the `null` case because its always nonsense, but I left it in for consistency with the `topics: Vec<OneOrMany<_>>` parameter.

Reply:
Makes sense, cheers for the explanation. If this is something API clients might feasibly expect then certainly makes sense to keep it.